packages/client/hmi-client/src/components/workflow/ops/stratify-mira/tera-stratify-node.vue

```diff
@@ -27,31 +27,56 @@ const props = defineProps<{
 	node: WorkflowNode<StratifyOperationStateMira>;
 }>();
 
+// /models/from-model-configuration/:config-id
+
 watch(
 	() => props.node.inputs,
 	async () => {
 		const input = props.node.inputs[0];
 		// Create a default if we dont have an output yet:
-		if (input && !props.node.outputs[0].value) {
-			let modelId: string | null = null;
+		// if (input && !props.node.outputs[0].value) {
+		if (input && input.value) {
+			let baseModelId = '';
+
 			if (input.type === 'modelId') {
-				modelId = input.value?.[0];
+				baseModelId = input.value[0];
 			} else if (input.type === 'modelConfigId') {
-				modelId = await getModelIdFromModelConfigurationId(input.value?.[0]);
+				const modelConfigId = input.value?.[0];
+
+				const modelConfiguration = await getModelConfigurationById(modelConfigId);
+				const model = (await getAsConfiguredModel(modelConfigId)) as Model;
+
+				// Mark the model as orginating from the config
+				model.temporary = true;
+				model.name += ` (${modelConfiguration.name})`;
+				model.header.name = model.name as string;
+
+				const res = await createModel(model);
+				if (res) {
+					baseModelId = res.id as string;
+				}
 			}
-			if (!modelId) return;
+			if (!baseModelId) return;
 
-			const model = await getModel(modelId);
+			const model = await getModel(baseModelId);
 			const modelName = model?.name;
-			emit('append-output', {
-				type: StratifyMiraOperation.outputs[0].type,
-				label: modelName ?? 'Default Model',
-				value: modelId,
-				state: {
-					strataGroup: cloneDeep(props.node.state.strataGroup),
-					strataCodeHistory: cloneDeep(props.node.state.strataCodeHistory)
-				}
-			});
+
+			const state = cloneDeep(props.node.state);
+			state.baseModelId = baseModelId;
+
+			emit(
+				'append-output',
+				{
+					type: StratifyMiraOperation.outputs[0].type,
+					label: modelName ?? 'Default Model',
+					value: baseModelId,
+					state: {
+						strataGroup: cloneDeep(props.node.state.strataGroup),
+						strataCodeHistory: cloneDeep(props.node.state.strataCodeHistory)
+					}
+				},
+				state
```

Comment:
Why do we send the state twice is some fashion?

Reply:
We will send output's state within that inner json and the last section here (line 78) will update the general state

Reply:
Thank you both for the explaination